presto-password-authenticators/src/test/java/io/prestosql/plugin/password/ldap/TestLdapAuthenticator.java

```diff
@@ -50,4 +144,67 @@ public void testContainsSpecialCharacters()
                 .as("Angle brackets")
                 .isEqualTo(true);
     }
+
+    private static class TestLdapAuthenticatorClient
```

Comment:
We also need real product tests with real ldap service. See `io.prestosql.tests.jdbc.TestLdapPrestoJdbc`

Reply:
I don't think we need a real product test.  This code verifies that both paths are checked.  So for an integration, if one test works, multiple will work by extension.